packages/firestore/src/remote/remote_event.ts

```diff
@@ -70,6 +70,70 @@ export class RemoteEvent {
       resumeToken: emptyByteString()
     };
   }
+
+  /**
+   * Strips out mapping changes that aren't actually changes. That is, if the document already
+   * existed in the target, and is being added in the target, and this is not a reset, we can
+   * skip doing the work to associate the document with the target because it has already been done.
+   */
+  filterUpdatesFromTargetChange(
+    targetChange: TargetChange,
+    existingDocuments: DocumentKeySet
+  ): void {
+    if (targetChange.mapping instanceof UpdateMapping) {
+      const update = targetChange.mapping;
+      const added = update.addedDocuments;
+      let result = added;
+      added.forEach(key => {
```

Comment:
Suggestion: We could also consider adding this functionality to SortedSet.

Reply:
I'm not opposed to this, but from a quick look, it appears we want a few classic collection methods: `filter`, `foreach`, and maybe `collect`?

This seems like the sort of thing we would want to do in sync across all of the `SortedSet` and `SortedMap` implementations. Maybe file a bug for it? I don't think it should be part of this change.

Reply:
That's fair, but I will ask you to do next time you implement `filter` :)